library/Jira/Web/RenderingHelper.php

```diff
@@ -17,49 +19,207 @@ class RenderingHelper
+            $url = Url::fromPath($match[2]);
+            $link = new Link($match[1], $url);
+
+            $monitoringObjectLink = strpos($match[2], 'icingaweb2/monitoring/host/show') !== false
+                || strpos($match[2], 'icingaweb2/monitoring/service/show') !== false;
+
+            $icingadbObjectLink = strpos($match[2], 'icingaweb2/icingadb/host') !== false
+                || strpos($match[2], 'icingaweb2/icingadb/service') !== false;
+
+            if ($monitoringObjectLink || $icingadbObjectLink) {
+                if (strpos($match[2], 'service') !== false) {
+                    if ($monitoringObjectLink) {
+                        $urlServiceParam = $url->getParam('service');
+                        $urlHostParam = $url->getParam('host');
+                        if (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()) {
+                            $link->setUrl(Url::fromPath(
+                                'icingadb/service',
+                                [
+                                    'name'      => $urlServiceParam,
+                                    'host.name' => $urlHostParam,
+                                ]
+                            ));
+                        }
+                    } else {
+                        $urlServiceParam = $url->getParam('name');
+                        $urlHostParam = $url->getParam('host.name');
+                    }
+
+                    if ($urlServiceParam === $this->serviceName && $urlHostParam === $this->hostName) {
+                        $serviceLink = clone $link;
+                        $serviceLink->setContent([new Icon('cog'), $match[1]])
+                            ->addAttributes(['title' => t('Show Icinga Service State')]);
+                        $this->setServiceLink($serviceLink);
+                    }
+                } else {
+                    $icon = new Icon('server');
+                    if ($monitoringObjectLink) {
+                        $urlHostParam = $url->getParam('host');
+                        if (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()) {
+                            $link->setUrl(Url::fromPath(
+                                'icingadb/host',
+                                ['name' => $urlHostParam]
+                            ));
+                        } else {
+                            $icon = new Icon('laptop');
+                        }
+                    } else {
+                        $urlHostParam = $url->getParam('name');
+                    }
+
+                    if ($urlHostParam === $this->hostName) {
+                        $hostLink = clone $link;
+                        $hostLink->setContent([$icon, $match[1]])
+                            ->addAttributes(['title' => t('Show Icinga Host State')]);
+                        $this->setHostLink($hostLink);
+                    }
+                }
```

Comment:
This has still numerous problems:

* `icingaweb2/` is **your** base URL, but that isn't always the case
  * Use `$url->getPath()` instead of `$match[2]`, it only returns the relative path
* Looking for `service` anywhere in the URL may lead to false positives
  * Use either `$url->getPath()` to search for `/service` or `$url->hasParam('service')` for a parameter
  * But don't transform `monitoring/host/show?host=docker-master&service=disk` to an Icinga DB Web service URL just because there's a `service` param
* This transforms any monitoring link now, not just the one that got automatically added. So there might be a chance that additional parameters are part of the URL. Overriding the link's URL with a completely new URL drops those.
  * If you transform from monitoring to Icinga DB Web, specifically override the path with `$url->setPath()`, remove old parameters with `$url->remove()` and add new ones with `$url->setParams()`